circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/DiffGeneratedPartitionPredicate.java

```diff
@@ -45,6 +46,7 @@
   private final Function<Path, String> checksumFunction;
   private String partitionPredicate;
   private boolean generated = false;
+  private Short partitionLimit = -1;
```

Comment:
How comes this isn't initialised to `UNLIMITED` here? This is only to figure out the limit on line 183 in `HiveDifferences` right?

Reply:
UNLIMITED is max int
partitionLimit for hive fetch partition call is -1 for being unlimited.
So I map -1 to max int in HiveDiferences so my loop nicely reads while (i < limit) https://github.com/HotelsDotCom/circus-train/blob/3a5e3bfaefe28ddcd500fa8a3451f6489005b2e7/circus-train-comparator/src/main/java/com/hotels/bdp/circustrain/comparator/hive/HiveDifferences.java#L219
 it's mostly sugarcoating to be honest...